src/main/java/fi/metatavu/metamind/bot/BotController.java

```diff
@@ -110,35 +111,50 @@ public MetamindBot getBotInstance(Story story) {
     String storyJson = story.getStoryJson();
     
     Map<String, SlotModel> slotModels = new HashMap<>();
-    IntentModel intentModel = null;
+    List<IntentModel> intentModels = new ArrayList<IntentModel>();
+    List<MachineLearningConfig> machineLearningConfigs = storyConfig.getMachineLearning();
     
     if (storyConfig.getMachineLearning() != null) {
       loadSlotModels(storyConfig, slotModels);
       
-      String intentModelName = storyConfig.getMachineLearning().getIntentModel();
-      if (intentModelName != null) {
-        intentModel = modelsContoller.findIntentModelByName(String.format("%s.bin", intentModelName));
-        if (intentModel == null && logger.isWarnEnabled()) {
-          logger.warn(String.format("Failed to load intent model %s", intentModelName));
+      List<String> intentModelNames = new ArrayList<String>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)